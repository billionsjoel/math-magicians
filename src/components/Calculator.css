.App {
	text-align: center;
}

.grid-container {
	display: grid;
	grid-template-columns: auto auto auto auto;
	background-color: #f1f4f7;
}

.grid-item {
	border: 0.2px solid #cac7c7;
	text-align: center;
	font-size: 20px;
	padding: 2rem;
}

.grid-item:hover {
	background-color: lime;
	cursor: pointer;
}

.item-1 {
	grid-column: 1/5;
	text-align: end;
	color: #fff;
	background-color: rgb(125, 131, 131);
}

.item-operations {
	background-color: orange;
}

.item-zero {
	grid-column: 1/3;
}
