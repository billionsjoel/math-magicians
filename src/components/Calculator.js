/* eslint-disable no-mixed-spaces-and-tabs */
/* eslint-disable no-tabs */
import './Calculator.css';
import React, { useState } from 'react';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [state, setState] = useState({
    total: '0',
    next: null,
    operation: null,
  });

  const handleClick = (e) => {
    if (!e.target.name) return;
    const { next, total, operation } = calculate(state, e.target.name);

    if (next === null && total === null) {
      setState({ next, total: '0', operation });
    } else {
      setState({ next, total, operation });
    }
  };

  const buttons = ['AC', '+/-', '%', 'รท', '7', '8', '9', 'x', '4', '5', '6', '-', '1', '2', '3', '+', '0', '.', '='];
  const operators = 'รทx-+=';
  const { total, next } = state;

  return (
    <div className="App">
      <div className="grid-container">
        {total ? (
          <div className="input input grid-item item-1 answer-field">
            {total}
          </div>
        ) : (
          <div className="input input grid-item item-1 answer-field">
            {next}
          </div>
        )}
        {buttons.map((btn) => (
          <button
            onClick={handleClick}
            type="button"
            key={btn}
            id={btn}
            name={btn}
            className={`grid-item ${operators.includes(btn) ? 'operator' : ''
            } ${btn === '0' ? 'item-zero' : ''} ${btn === 'รท'
								|| btn === 'x'
								|| btn === '-'
								|| btn === '+'
								|| btn === '='
						  ? 'item-operations'
						  : ''
            }`}
          >
            {btn}
          </button>
        ))}
      </div>
    </div>
  );
};

export default Calculator;
