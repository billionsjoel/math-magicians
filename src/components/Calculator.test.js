import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from './Calculator';


