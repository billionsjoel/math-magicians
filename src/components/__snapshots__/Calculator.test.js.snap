// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders buttons 1`] = `
<div
  className="App"
>
  <div
    className="grid-container"
  >
    <div
      className="input input grid-item item-1 answer-field"
    >
      0
    </div>
    <button
      className="grid-item   "
      id="AC"
      name="AC"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="grid-item   "
      id="+/-"
      name="+/-"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="grid-item   "
      id="%"
      name="%"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="grid-item operator  item-operations"
      id="รท"
      name="รท"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="grid-item   "
      id="7"
      name="7"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="grid-item   "
      id="8"
      name="8"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="grid-item   "
      id="9"
      name="9"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="grid-item operator  item-operations"
      id="x"
      name="x"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="grid-item   "
      id="4"
      name="4"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="grid-item   "
      id="5"
      name="5"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="grid-item   "
      id="6"
      name="6"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="grid-item operator  item-operations"
      id="-"
      name="-"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="grid-item   "
      id="1"
      name="1"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="grid-item   "
      id="2"
      name="2"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="grid-item   "
      id="3"
      name="3"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="grid-item operator  item-operations"
      id="+"
      name="+"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="grid-item  item-zero "
      id="0"
      name="0"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="grid-item   "
      id="."
      name="."
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="grid-item operator  item-operations"
      id="="
      name="="
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
